From 861ca3d6787d84c9bea7110cb46821e9b8f63aff Mon Sep 17 00:00:00 2001
From: Sebastian Rasmussen <sebras@hotmail.com>
Date: Thu, 24 Jul 2014 22:02:58 +0200
Subject: [PATCH 15/15] parse: Unref reference to enclosing bins

Previously all reference to enclosing bins of an element were leaked
when doing delaying setting a property.

Fixes https://bugzilla.gnome.org/show_bug.cgi?id=733697
---
 gst/parse/grammar.y | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gst/parse/grammar.y b/gst/parse/grammar.y
index f5333897b..5bcd0dfe6 100644
--- a/gst/parse/grammar.y
+++ b/gst/parse/grammar.y
@@ -318,9 +318,12 @@ gst_parse_add_delayed_set (GstElement *element, gchar *name, gchar *value_str)
         gst_parse_add_delayed_set(parent, sub_name, value_str);
         g_free (sub_name);
       }
+      gst_object_unref (parent);
       parent = child;
       current++;
     }
+    if (parent)
+      gst_object_unref (parent);
     g_strfreev (names);
   }
 }
-- 
2.19.0

